File: src/main/kotlin/org/kite9/diagram/common/elements/Positioned.java
package org.kite9.diagram.common.elements;

import org.kite9.diagram.model.position.Dimension2D;

public interface Positioned {

	public void setX(double x);

	public void setY(double y);

	public double getX();

	public double getY();

	public Dimension2D getPosition();
}

File: kite9-visualization/src/commonMain/kotlin/org/kite9/diagram/model/Positioned.kt
package org.kite9.diagram.model

import org.kite9.diagram.common.elements.Dimension
import org.kite9.diagram.model.position.RectangleRenderingInformation
import org.kite9.diagram.model.style.RectangularPosition


/**
 * Positioned elements have an x,y position within a diagram.  If they also have width and height
 * they will be represented by a subclass of this, Rectangular.
 */
interface Positioned : DiagramElement {

    override fun getRenderingInformation(): RectangleRenderingInformation

    /**
     * Any other details about how this is to be positioned in its container, in a given
     * dimension.
     */
    fun getContainerPosition(d: Dimension): RectangularPosition?
}